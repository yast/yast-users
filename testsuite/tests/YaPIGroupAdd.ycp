/**
 * File		: YaPIGroupAdd.ycp
 * Module	: Users configurator
 * Summary	: Test of USERS::GroupAdd function
 * Author	: Jiri Suchomel <jsuchome@suse.cz>
 *
 * $Id$
 */

{
    // testedfiles: Users.pm UsersCache.pm USERS.pm


    import "Directory";
    import "Mode";
    import "Users";
    import "UsersPasswd";
    import "YaPI::USERS";

    string tmpdir	= Directory::tmpdir;
    foreach (string file, [ "passwd", "group", "shadow" ], {
	SCR::Execute (.target.bash, sformat ("/bin/cp ./%1 %2/", file, tmpdir));
    });
    Users::SetBaseDirectory (tmpdir);
    Users::ReadLocal ();

    import "Testsuite";

    map R = $[
	"etc" : $[
	    "fstab": [],
	    "cryptotab": [],
	    "default": $[
		"useradd": $[
		    "home":	"/tmp/local/home",
		    "groups":	"audio,video",
		    "expire":	nil,
		    "group":	100
		]
	    ],
	],
	"target": $[
	    "stat"	: $[],
	    "size"	: -1,
	    "tmpdir"	: "/tmp/YaST"
	],
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "IO_SCHEDULER"		: "",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
    ];
    map W = $[];
    map E = $[
	"passwd" : $[
	    "init"	: true
	],
	"target": $[
	    "mkdir"	: true,
	    "bash"	: 0,
	    "bash_output": $[],
	]
    ];

    Testsuite::Dump ("==========================================================");

    Mode::SetTest ("test");

    Testsuite::Test (``(Users::SetBaseDirectory ("/etc")), [], 0);
    Testsuite::Test (``(UsersPasswd::SetBaseDirectory ("/etc")), [], 0);

    map<string,any> config_map  = $[];
    map<string,any> data_map    = $[
	"cn"	: "gg",
    ];
    E ["target", "bash_output", "stdout"] = "gg";

    Testsuite::Dump ("============ add new group 'gg': ==========================");
    Testsuite::Test (``(YaPI::USERS::GroupAdd (config_map, data_map)), [R,W,E], 0);
    Testsuite::Dump ("============ add new group 'gg' - done ====================");


    data_map ["userlist"]	= [ "hh" ];
    data_map ["cn"]		= "gg2";
    E ["target", "bash_output", "stdout"] = "gg2";

    Testsuite::Dump ("============ add new group 'gg2' with first userlist as list ==");
    Testsuite::Test (``(YaPI::USERS::GroupAdd (config_map, data_map)), [R,W,E], 0);
    Testsuite::Dump ("============ add new group 'gg2' - done ====================");

    data_map ["userlist"]	= $[
	"hh"	: 1,
	"ii"	: 1
    ];
    data_map ["cn"]		= "gg3";
    E ["target", "bash_output", "stdout"] = "gg3";

    Testsuite::Dump ("============ add new group 'gg3' with first userlist as map ==");
    Testsuite::Test (``(YaPI::USERS::GroupAdd (config_map, data_map)), [R,W,E], 0);
    Testsuite::Dump ("============ add new group 'gg3' - done ====================");

    data_map ["userlist"]	= $[
	"hh2"	: 1,
    ];
    data_map ["cn"]		= "gg4";
    E ["target", "bash_output", "stdout"] = "gg4";

    Testsuite::Dump ("============ add new group 'gg4' with non existent user ==");
    Testsuite::Test (``(YaPI::USERS::GroupAdd (config_map, data_map)), [R,W,E], 0);
    Testsuite::Dump ("============ add new group 'gg4' - done ====================");

    Testsuite::Dump ("============ add new group 'root' (groupname conflict): ======");

    data_map	= $[
	"cn"	: "root"
    ];
    E ["target", "bash_output", "stdout"] = "root";

    string error = (string)
	Testsuite::Test (``(YaPI::USERS::GroupAdd (config_map, data_map)), [R,W,E], 0);
    Testsuite::Dump (sformat ("------------ GroupAdd return value:\n%1", error));

    Testsuite::Dump ("============ add new group 'root' - done ====================");
    Testsuite::Dump ("==========================================================");
}
