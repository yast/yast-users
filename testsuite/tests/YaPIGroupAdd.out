Dump	==========================================================
Return	nil
Return	nil
Dump	============ add new group 'gg': ==========================
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/tmp/local/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.target.bash_output "echo 'gg' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"gg"]
Execute	.target.bash "/bin/cp /etc/group /etc/group.YaST2save" 0
Write	.target.string "/etc/group" "audio:x:17:ii\nnobody:x:65533:\nroot:x:0:\ngg:x:1000:\nusers:x:100:\n+:::\n" true
Execute	.target.bash "/usr/sbin/nscd -i group" 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $["custom_groups":["local"], "custom_users":["local"], "dont_warn_when_nisserver_notdes":false, "dont_warn_when_uppercase":false] true
Return	
Dump	============ add new group 'gg' - done ====================
Dump	============ add new group 'gg2' with first userlist as list ==
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/tmp/local/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.target.bash_output "echo 'gg2' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"gg2"]
Execute	.target.bash "/bin/cp /etc/group /etc/group.YaST2save" 0
Write	.target.string "/etc/group" "audio:x:17:ii\nnobody:x:65533:\nroot:x:0:\ngg:x:1000:\ngg2:x:1001:hh\nusers:x:100:\n+:::\n" true
Execute	.target.bash "/usr/sbin/nscd -i group" 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $["custom_groups":["local"], "custom_users":["local"], "dont_warn_when_nisserver_notdes":false, "dont_warn_when_uppercase":false] true
Return	
Dump	============ add new group 'gg2' - done ====================
Dump	============ add new group 'gg3' with first userlist as map ==
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/tmp/local/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.target.bash_output "echo 'gg3' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"gg3"]
Execute	.target.bash "/bin/cp /etc/group /etc/group.YaST2save" 0
Write	.target.string "/etc/group" "audio:x:17:ii\nnobody:x:65533:\nroot:x:0:\ngg:x:1000:\ngg2:x:1001:hh\ngg3:x:1002:hh,ii\nusers:x:100:\n+:::\n" true
Execute	.target.bash "/usr/sbin/nscd -i group" 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $["custom_groups":["local"], "custom_users":["local"], "dont_warn_when_nisserver_notdes":false, "dont_warn_when_uppercase":false] true
Return	
Dump	============ add new group 'gg3' - done ====================
Dump	============ add new group 'gg4' with non existent user ==
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/tmp/local/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.target.bash_output "echo 'gg4' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"gg4"]
Return	User hh2 does not exist.
Dump	============ add new group 'gg4' - done ====================
Dump	============ add new group 'root' (groupname conflict): ======
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/tmp/local/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.target.bash_output "echo 'root' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"root"]
Return	There is a conflict between the entered
Dump	------------ GroupAdd return value:
Dump	There is a conflict between the entered
Dump	group name and an existing group name.
Dump	Try another one.
Dump	============ add new group 'root' - done ====================
Dump	==========================================================
